### ISSUE TITLE
no check if need causal mask in flash_attn path

### ISSUE THREAD

**Issue by inspirit** [lucidrains/soundstorm-pytorch#37]
Hi Phil,
Looks like you forgot to check if actually need to apply causal mask to `attn_bias` when using flash attention path:
https://github.com/lucidrains/soundstorm-pytorch/blob/d4830b43355a314a55afe6dc2cf9b3ba4fc9bcc5/soundstorm_pytorch/attend.py#L98

**Comment by lucidrains** [comment]
@inspirit :man_facepalming: yes, thank you

**Comment by lucidrains** [comment]
@inspirit have you seen https://github.com/lucidrains/e2-tts-pytorch ? think i can revisit soundstorm and add the E2 type conditioning here as well (and cleanup a bunch of old torch code along the way)

**Comment by inspirit** [comment]
yeah that one is of particular interest for me as well :) i was planning on giving it a try soon. I tried another implementation of e2-tts (rfwave) and it worked quite well for me but requires a lot of training, so i will come back to e2-tts a bit later.

i'm currently trying to apply maskgit generator to RQVAE from different domain so needs more adjusting here and there

**Comment by lucidrains** [comment]
yes that's a good combo
